src/providers/ibmcloud/mod.rs

```diff
@@ -112,6 +126,29 @@ impl IBMGen2Provider {
         }
         output
     }
+
+    /// Find the SSH keys in the vendordata file
+    fn fetch_ssh_keys(vendordata_vec: Vec<u8>) -> Result<Vec<String>> {
+        // Parse MIME format from vendor-data file
+        let vendor_data_mail = parse_mail(&vendordata_vec).unwrap();
+        let mut cloud_config = String::new();
+        for section in vendor_data_mail.subparts {
+            for header in &section.headers {
+                if let "text/cloud-config" = header.get_value().as_str() {
+                    if section.get_body().unwrap().contains("ssh_authorized_keys") {
+                        cloud_config = section
+                            .get_body()
+                            .chain_err(|| "failed to get cloud-config content")?;
+                        break;
+                    }
+                }
+            }
+        }
+        // Parse YAML to find SSH keys
+        let deserialized_cloud_config: VendorDataCloudConfig = serde_yaml::from_str(&cloud_config)
```

Comment:
[Minor UX concern]
There is a chance that `cloud_config` is still an empty string at this point.
I think it would be good to have an `is_empty()` check before this and return an helpful error (something like "no cloud-config section found in vendor-data") before trying to deserialize.

Reply:
Added this check in the next commit.